### PR TITLE
Add py.typed

### PR DESCRIPTION
**Pull request by twoertwein**
`ABCExtensionArray` and `ABCSeries` did not exist in the stubs but this error might have been hidden because the pandas implementation provides it.

### CODE REVIEW

File: pandas-stubs/core/groupby/groupby.pyi [hunk 1]
```diff
@@ -7,7 +7,6 @@ from pandas.core.base import (
 from pandas.core.frame import DataFrame
 from pandas.core.generic import NDFrame
 from pandas.core.groupby import ops
-from pandas.core.groupby.indexing import GroupByIndexingMixin
```
Comment (twoertwein) [inline pandas-stubs/core/groupby/groupby.pyi:10]: not defined in the stubs - but it provides only private methods, so I removed it.

**Comment by twoertwein** [comment]
A much easier fix would be to simply have `pandas-stubs/py.typed` then we don't need renaming.

@Dr-Irv I now you removed it at some point because of a bug(?) in pyright. Is it now safe to enable it again?

**Comment by Dr-Irv** [comment]
> A much easier fix would be to simply have `pandas-stubs/py.typed` then we don't need renaming.
> 
> @Dr-Irv I now you removed it at some point because of a bug(?) in pyright. Is it now safe to enable it again?

Don't know.  If that would work, I'd prefer it over having to rename `pandas` during testing.  Can you try that?

**Comment by twoertwein** [comment]
> Don't know. If that would work, I'd prefer it over having to rename `pandas` during testing. Can you try that?

If I remove `ABCExtensionArray` and just have a py.typed, mypy/pyright notice the error :)

File: pandas-stubs/core/dtypes/generic.pyi [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from pandas.core.arrays import ExtensionArray
+from pandas.core.generic import NDFrame
+
+ABCIndex = type[Index]
+
+ABCNDFrame = type[NDFrame]
+ABCSeries = type[Series]
+ABCDataFrame = type[DataFrame]
+
+ABCExtensionArray = type[ExtensionArray]
```
Comment (twoertwein) [inline pandas-stubs/core/dtypes/generic.pyi:15]: Technically need only `ABCExtensionArray` and `ABCSeries` for the current stubs.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
I think we should have `py.typed` with the word `partial` in it.

**Comment by twoertwein** [comment]
https://peps.python.org/pep-0561/
> Many stub packages will only have part of the type interface for libraries completed, especially initially. For the benefit of type checking and code editors, packages can be “partial”. This means modules not found in the stub package SHOULD be searched for in parts four and five of the module resolution order above, namely inline packages and typeshed.


If we add `partial`, mypy does not find the missing `ABCExtensionArray` (and users's type checkers will also fallback on the pandas implementation). While pandas-stubs is not complete, I would slightly favor marking it as complete, so that people report any missing annotations to make pandas-stub more complete.

**Comment by Dr-Irv** [comment]
> If we add `partial`, mypy does not find the missing `ABCExtensionArray` (and users's type checkers will also fallback on the pandas implementation). While pandas-stubs is not complete, I would slightly favor marking it as complete, so that people report any missing annotations to make pandas-stub more complete.

OK, I'm fine with that.

**Review by Dr-Irv — APPROVED** [review]
thanks @twoertwein

**Pull request merged**